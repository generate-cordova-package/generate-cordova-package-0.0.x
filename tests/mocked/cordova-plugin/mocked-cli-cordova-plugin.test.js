/* global jest, it, expect */
/* (see <https://github.com/standard/standard/issues/905>) */

const snapshotDiff = require('snapshot-diff')

const startMain = require('../../../lib/start-main-cli')

const renderCordovaPluginTree = require('../../../lib/cordova-plugin/render')

const renderCordovaPluginProperties = require('../../render-cordova-plugin/render-cordova-plugin-properties.config')

const mockOutputLog = []

const mockMainPromptResponse = {
  moduleType: 'cordova-plugin'
}

const mockPluginResponseProperties = renderCordovaPluginProperties

jest.mock('gitconfig', () => ({
  get: options => {
    mockOutputLog.push({ gitconfig: options })

    return Promise.resolve({
      user: { name: 'chris', email: 'chris@chris.com' }
    })
  }
}))

jest.mock('prompts', () => (questions, _) => {
  // with options ignored in this test case

  mockOutputLog.push({ prompts: { questions } })

  return Promise.resolve(
    questions[0].name === 'moduleType'
      ? mockMainPromptResponse
      : mockPluginResponseProperties
  )
})

jest.mock('moment', () => () => ({
  // (starting to look forward):
  year: () => 2023
}))

jest.mock('fs-tree', () => tree => {
  mockOutputLog.push({ fsTree: tree })
})

it('start and run mocked CLI to create cordova-plugin', async () => {
  await startMain()

  // NOTE that the results of `lib/cordova-plugin/render.js`
  // are checked in another test case;
  // no need to check here.
  expect(
    snapshotDiff(
      // with rendered tree in an array to make snapshot-diff happy
      [renderCordovaPluginTree(renderCordovaPluginProperties)],
      mockOutputLog,
      // cleaner snapshot diffs
      { contextLines: 1, stablePatchmarks: true }
    )
  ).toMatchSnapshot()
})
