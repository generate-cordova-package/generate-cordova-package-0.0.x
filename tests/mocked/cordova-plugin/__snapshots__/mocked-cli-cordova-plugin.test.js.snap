// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`start and run mocked CLI to create cordova-plugin 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ --- --- @@
    Object {
+     \\"prompts\\": Object {
+       \\"questions\\": Array [
+         Object {
+           \\"choices\\": Array [
+             Object {
+               \\"title\\": \\"Cordova app\\",
+               \\"value\\": \\"cordova-app\\",
+             },
+             Object {
+               \\"title\\": \\"Cordova plugin\\",
+               \\"value\\": \\"cordova-plugin\\",
+             },
+           ],
+           \\"default\\": 0,
+           \\"initial\\": 0,
+           \\"message\\": \\"Select module type\\",
+           \\"name\\": \\"moduleType\\",
+           \\"type\\": \\"select\\",
+         },
+       ],
+     },
+   },
+   Object {
+     \\"gitconfig\\": Object {
+       \\"location\\": \\"global\\",
+     },
+   },
+   Object {
+     \\"prompts\\": Object {
+       \\"questions\\": Array [
+         Object {
+           \\"default\\": \\"demo-plugin\\",
+           \\"initial\\": \\"demo-plugin\\",
+           \\"message\\": \\"Cordova plugin name\\",
+           \\"name\\": \\"name\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"com.demoplugin\\",
+           \\"initial\\": \\"com.demoplugin\\",
+           \\"message\\": \\"pluin id\\",
+           \\"name\\": \\"id\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"demo plugin\\",
+           \\"initial\\": \\"demo plugin\\",
+           \\"message\\": \\"Cordova plugin description\\",
+           \\"name\\": \\"description\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"0.0.1-dev\\",
+           \\"initial\\": \\"0.0.1-dev\\",
+           \\"message\\": \\"Initial plugin version\\",
+           \\"name\\": \\"version\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"MIT\\",
+           \\"initial\\": \\"MIT\\",
+           \\"message\\": \\"Cordova plugin license\\",
+           \\"name\\": \\"license\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"chris\\",
+           \\"initial\\": \\"chris\\",
+           \\"message\\": \\"Author name\\",
+           \\"name\\": \\"author\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"chris@chris.com\\",
+           \\"initial\\": \\"chris@chris.com\\",
+           \\"message\\": \\"Author email\\",
+           \\"name\\": \\"email\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"demo\\",
+           \\"initial\\": \\"demo\\",
+           \\"message\\": \\"plugin JavaScript module name (without js extension)\\",
+           \\"name\\": \\"jsModuleName\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"Demo\\",
+           \\"initial\\": \\"Demo\\",
+           \\"message\\": \\"plugin JavaScript object name\\",
+           \\"name\\": \\"jsObjectName\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"demo\\",
+           \\"initial\\": \\"demo\\",
+           \\"message\\": \\"plugin JavaScript function name\\",
+           \\"name\\": \\"jsFunctionName\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"Demo\\",
+           \\"initial\\": \\"Demo\\",
+           \\"message\\": \\"native object name\\",
+           \\"name\\": \\"nativeObjectName\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"com.demo\\",
+           \\"initial\\": \\"com.demo\\",
+           \\"message\\": \\"Android package name\\",
+           \\"name\\": \\"androidPackageName\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"demo\\",
+           \\"initial\\": \\"demo\\",
+           \\"message\\": \\"native function name\\",
+           \\"name\\": \\"nativeFunctionName\\",
+           \\"type\\": \\"text\\",
+         },
+       ],
+     },
+   },
+   Object {
+     \\"fsTree\\": Object {
        \\"test-plugin\\": Object {
@@ --- --- @@
      },
+   },
    Object {
-     \\"undefined\\": Object {
-       \\"README.md\\": \\"# 
+     \\"prompts\\": Object {
+       \\"questions\\": Array [
+         Object {
+           \\"default\\": \\"demo-app\\",
+           \\"initial\\": \\"demo-app\\",
+           \\"message\\": \\"Cordova plugin demo app name\\",
+           \\"name\\": \\"demoAppName\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"com.demoapp\\",
+           \\"initial\\": \\"com.demoapp\\",
+           \\"message\\": \\"Cordova plugin demo app id\\",
+           \\"name\\": \\"id\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"demo app\\",
+           \\"initial\\": \\"demo app\\",
+           \\"message\\": \\"Cordova plugin demo app description\\",
+           \\"name\\": \\"description\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"0.0.1-dev\\",
+           \\"initial\\": \\"0.0.1-dev\\",
+           \\"message\\": \\"plugin demo app version\\",
+           \\"name\\": \\"version\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"MIT\\",
+           \\"initial\\": \\"MIT\\",
+           \\"message\\": \\"plugin demo app license\\",
+           \\"name\\": \\"license\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"Alice\\",
+           \\"initial\\": \\"Alice\\",
+           \\"message\\": \\"demo app author name\\",
+           \\"name\\": \\"author\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"alice@alice.com\\",
+           \\"initial\\": \\"alice@alice.com\\",
+           \\"message\\": \\"demo app author email\\",
+           \\"name\\": \\"email\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"TestObject\\",
+           \\"initial\\": \\"TestObject\\",
+           \\"message\\": \\"demo app JavaScript object name from plugin (should match plugin)\\",
+           \\"name\\": \\"jsObjectName\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"testFunction\\",
+           \\"initial\\": \\"testFunction\\",
+           \\"message\\": \\"demo app JavaScript function name from plugin (should match plugin)\\",
+           \\"name\\": \\"jsFunctionName\\",
+           \\"type\\": \\"text\\",
+         },
+       ],
+     },
+   },
+   Object {
+     \\"fsTree\\": Object {
+       \\"test-plugin-demo-app\\": Object {
+         \\"README.md\\": \\"# test-plugin-demo-app

@@ --- --- @@
  \\",
-       \\"config.xml\\": \\"<widget id=\\\\\\"cc.testplugin\\\\\\" version=\\\\\\"0.1.1\\\\\\">
+         \\"config.xml\\": \\"<widget id=\\\\\\"cc.plugintestdemo\\\\\\" version=\\\\\\"0.1.1\\\\\\">
+   <name>test-plugin-demo-app</name>
    <description>test plugin</description>
@@ --- --- @@
          \\"package.json\\": \\"{
+   \\\\\\"name\\\\\\": \\\\\\"test-plugin-demo-app\\\\\\",
    \\\\\\"version\\\\\\": \\\\\\"0.1.1\\\\\\",
@@ --- --- @@
    <body>
+     <h1>test-plugin-demo-app</h1>
      <div id=\\\\\\"status\\\\\\"/>
@@ --- --- @@
  \\",
+         },
        },"
`;
